From 460a27eb4530d9d7f7eceaf42b1f1b13af96d952 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Thu, 11 Jun 2020 10:29:09 +0800
Subject: [PATCH] rabbitmq-script-wrapper: allow to run as root

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 scripts/rabbitmq-script-wrapper | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/scripts/rabbitmq-script-wrapper b/scripts/rabbitmq-script-wrapper
index 9623f01..b2a4520 100644
--- a/scripts/rabbitmq-script-wrapper
+++ b/scripts/rabbitmq-script-wrapper
@@ -15,9 +15,14 @@
 ##  Copyright (c) 2007-2015 Pivotal Software, Inc.  All rights reserved.
 ##
 
+SED_OPT="-E"
+if [ $(uname -s) = "Linux" ]; then
+    SED_OPT="-r"
+fi
+
 for arg in "$@" ; do
     # Wrap each arg in single quotes and wrap single quotes in double quotes, so that they're passed through cleanly.
-    arg=`printf %s "$arg" | sed -e "s#'#'\"'\"'#g"`
+    arg=`printf %s "$arg" | sed $SED_OPT -e "s/'/'\"'\"'/g"`
     CMDLINE="${CMDLINE} '${arg}'"
 done
 
@@ -30,14 +35,16 @@ if [ `id -u` = `id -u rabbitmq` -a "$SCRIPT" = "rabbitmq-server" ] ; then
     RABBITMQ_SCRIPTS_DIR=$(dirname "$RABBITMQ_ENV")
     . "$RABBITMQ_ENV"
 
-    exec /usr/lib/rabbitmq/bin/rabbitmq-server "$@" @STDOUT_STDERR_REDIRECTION@
+    exec /usr/lib/rabbitmq/bin/rabbitmq-server "$@"
 elif [ `id -u` = `id -u rabbitmq` -o "$SCRIPT" = "rabbitmq-plugins" ] ; then
     if [ -f $PWD/.erlang.cookie ] ; then
         export HOME=.
     fi
     exec /usr/lib/rabbitmq/bin/${SCRIPT} "$@"
 elif [ `id -u` = 0 ] ; then
-    @SU_RABBITMQ_SH_C@ "/usr/lib/rabbitmq/bin/${SCRIPT} ${CMDLINE}"
+    # WRS. Allow to run as root
+    export HOME=${HOME:-/root}
+    /bin/sh -c "/usr/lib/rabbitmq/bin/${SCRIPT} ${CMDLINE}"
 else
     /usr/lib/rabbitmq/bin/${SCRIPT}
     echo
-- 
2.7.4

